tests/src/test/java/alluxio/master/lineage/LineageMasterIntegrationTest.java

```diff
@@ -150,21 +152,25 @@ public void lineageRecoveryTest() throws Exception {
       out.write("foo".getBytes());
       out.close();
       lineageClient.reportLostFile("/testFile");
-      // Wait for the log file to be created by the recompute job
-      while (!logFile.exists()) {
-        CommonUtils.sleepMs(20);
-      }
-      // Wait for the output to be written (this should be very fast)
-      CommonUtils.sleepMs(10);
-      BufferedReader reader = new BufferedReader(new FileReader(logFile));
-      try {
-        Assert.assertEquals("hello world", reader.readLine());
-      } finally {
-        reader.close();
-      }
     } finally {
       lineageClient.close();
     }
+
+    // Wait for the log file to be created by the recompute job
+    CommonTestUtils.waitFor("the log file to be written", new Function<Void, Boolean>() {
+      @Override
+      public Boolean apply(Void input) {
+        if (!logFile.exists()) {
+          return false;
+        }
+        try (BufferedReader reader = new BufferedReader(new FileReader(logFile))) {
```

Comment:
It retries now until it finds what it's looking for, whereas before it would only try once.